source/lib/buffer.h

```diff
@@ -0,0 +1,52 @@
+#ifndef _BUFFER_H_
+#define _BUFFER_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "message_types.h"
+#include "FreeRTOS.h"
+#include "semphr.h"
+
+// NOTE: if you make this value larger, you have to update the types for front and back in the c_buffer
+// you also have to update how we update the front and end pointers when doing operations
+#define MESSAGE_BUFFER_SIZE 256
+
+/**
+ * A structure for storing messages. On overflow, will push the front pointer forwards.
+ * NOTE: If you want to do multiple operations, such as check the size and then push/pop,
+ * it is safe to acquire the lock outside of push/pop and it will not result in deadlock.
+*/
```

Comment:
If we end up doing this with any reasonable frequency, we can add some functions to check this to this file. Otherwise this is a great precaution to have, I think.